core/predicate_check.go

```diff
@@ -0,0 +1,41 @@
+package core
+
+import (
+	"github.com/ava-labs/avalanchego/snow"
+	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/utils"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+// CheckPredicatesForSenderTxs checks the stateful precompile predicates of any of the given
+// transactions that reference a stateful precompile address in their access list.
+// The parameter [txs] represents a flattened, nonce-ordered list of transactions originating from the same sender
+// Returns [len(txs), nil] if and only if all referenced predicates are met.
+// Otherwise, returns the index of the first transaction that was invalid and the predicate error.
+// In the failure case, all transactions after and including the index that failed the predicate should be considered invalid,
+// since the input [txs] are nonce-ordered and from the same sender.
+func CheckPredicatesForSenderTxs(rules params.Rules, snowCtx *snow.Context, txs types.Transactions, proposerVMBlockCtx *block.Context) (int, error) {
```

Comment:
nit: can we change the order of parameters to `rules, snowCtx, proposerVMBlockCtx, txs` this seems more cohesive to me